contrib/drivers/openstack/cinder/cinder.go

```diff
@@ -381,40 +414,52 @@ func (d *Driver) DeleteSnapshot(req *pb.DeleteVolumeSnapshotOpts) error {
 
 // ListPools
 func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
-	log.Info("Starting list pools in cinder drivers.")
-	opts := &schedulerstats.ListOpts{Detail: true}
-
-	pages, err := schedulerstats.List(d.blockStoragev2, opts).AllPages()
+	data, err := ioutil.ReadFile(defaultConfPath)
 	if err != nil {
-		log.Error("Cannot list storage pools:", err)
-		return nil, err
+		log.Error("Parse cinder.yaml fail!")
 	}
 
-	polpages, err := schedulerstats.ExtractStoragePools(pages)
-	if err != nil {
-		log.Error("Cannot extract storage pools:", err)
-		return nil, err
+	var typeName []string
+	ymlconf := CinderConfig{}
+	yaml.Unmarshal(data, &ymlconf)
+	for name, _ := range ymlconf.Pool {
+		typeName = append(typeName, name)
 	}
-	var pols []*model.StoragePoolSpec
-	for _, page := range polpages {
-		if _, ok := d.conf.Pool[page.Name]; !ok {
-			continue
+
+	var typePol []*model.StoragePoolSpec
+	for _, typRes := range typeName {
+		opts := ListPoolOpts{Detail: true, Volume_Type: typRes}
+		pages, err := schedulerstats.List(d.blockStoragev2, opts).AllPages()
+		if err != nil {
+			log.Error("Cannot list storage pools:", err)
+			return nil, err
 		}
 
-		pol := &model.StoragePoolSpec{
+		polpage, err := json.Marshal(pages.GetBody())
+		if err != nil {
+			log.Error("Get Storage body fail.")
+			return nil, err
+		}
+		var StoragePools PoolArray
+		json.Unmarshal(polpage, &StoragePools)
+		var freeCaps int64 = 0
+		var totalCaps int64 = 0
+		var pol *model.StoragePoolSpec
+		for _, page := range StoragePools.Pools {
+			freeCaps = freeCaps + int64(page.Capabilities.FreeCapacityGB)
+			totalCaps = totalCaps + int64(page.Capabilities.TotalCapacityGB)
```

Comment:
It is int64 here but it is FreeCapacityGB  float64 `json:"free_capacity_gb"` in cinder/cinder.go?